framework/tlvf/AutoGenerated/src/CmduMessageTx.cpp

```diff
@@ -84,11 +86,23 @@ bool CmduMessageTx::finalize(bool swap_needed)
     if (!m_cmdu_header)
         return false; 
 
-    if (m_cmdu_header->message_type() == eMessageType::VENDOR_SPECIFIC_MESSAGE &&
-        getClassCount() > 1) {
-        tlvVendorSpecific *tlv = dynamic_cast<tlvVendorSpecific *>(getClass(0).get());
-        if (tlv != nullptr) {
-            tlv->length() = getMessageLength() - (kCmduHeaderLength + kTlvHeaderLength);
+    // Update vendor specific length which is seperated to 3 classes
+    for (auto class_it = m_class_vector.begin(); class_it != m_class_vector.end(); class_it++) {
+        // Type is the first byte in the tlv buffer
+        uint8_t tlv_type = *(*class_it)->getStartBuffPtr();
+        if (tlv_type == uint8_t(eTlvType::TLV_VENDOR_SPECIFIC)) {
+            uint16_t* tlv_length = reinterpret_cast<uint16_t *>
+                ((class_it->get())->getStartBuffPtr() + sizeof(uint8_t));
+            // tlv vendor specific length already include OUI and beerocks header length
+            // need to add only beerocks message length.
+            // Classes order: +0 = tlvVendorSpecific, +1 = beerocks_header, +2 = beerocks_message
```

Comment:
I haven't identified why, but if I apply just this commit, it breaks. When backhaul manager goes to operational, it report:

backhaul_manager: TLV end of message not found! tlv_type=0, tlv_length=0, cmdu_type=9

But looking at the code: add_intel_vs_data() already updates the `tlv->length()`, so is this commit really needed?

Reply:
please see: https://github.com/prplfoundation/prplMesh/pull/321#discussion_r328700943